### PR TITLE
Add groups.SitePackagesGroup

### PR DESCRIPTION
**Pull request by PamelaM**
Also refactored some of utils.py for DRY.

TODO:
- [x] Add test against test_data/input.txt
- [x] Should the default configuration include "sitepackages"?  [It's part of PEP8, so I decided "yes"]
- [x] Identify and fix the `make lint` errors in files not changed by this PR [Needs separate PR]

### CODE REVIEW

**Comment by PamelaM** [comment]
Still need to update documentation

**Comment by miki725** [comment]
thanks @PamelaM!

looks like some flake8 is failing. let me know if you would rather merge and then I can fix violations or if you would like to help with that, that will be awesome as well.

**Comment by PamelaM** [comment]
I've fixed some of the flake8 failures, and I've also found and fixed a new bug in utls.site_pacakges_paths() where it wasn't putting sys.path back if there was an exception in the yield - which would cascade into later tests.

File: importanize/utils.py [hunk 1]
```diff
@@ -8,42 +8,57 @@
...
 
-def is_std_lib(module):
-    if not module:
+def _is_installed_module(module_name, filter_func, builtin_result):
+    if not module_name:
         return False
 
-    if module in sys.builtin_module_names:
-        return True
+    if module_name in sys.builtin_module_names:
```
Comment (PamelaM) [inline importanize/utils.py]: Changed `module` to `module_name` to disambiguate from actual "modules"

File: tests/test_utils.py [hunk 2]
```diff
@@ -7,26 +7,45 @@
...
-        with ignore_site_packages_paths():
+        with site_packages_paths(lambda i: "site-packages" not in i):
             self.assertNotEqual(sys.path, paths)
             self.assertLess(len(sys.path), len(paths))
 
         self.assertIn(os.getcwd(), sys.path)
         self.assertListEqual(sys.path, paths)
         sys.path.remove(os.getcwd())
 
+    def test_site_packages_paths_exception(self):
```
Comment (PamelaM) [inline tests/test_utils.py]: Q: should I DRY refactor this into self._test_site_packages_paths(raise_exception=False)?

**Comment by PamelaM** [comment]
Let's hold off on this, I've found a significant bug and will need to think about it.  I'll get back to it later.

File: tests/test_utils.py (hunk 2 above)
Comment (miki725) [inline tests/test_utils.py]: up to you 😄 

I usually like to leave old tests so that I have some confidence new refactor does not break old stuff.

**Comment by miki725** [comment]
curious, what is the bug?

**Comment by PamelaM** [comment]
The site-packages check is removing the stdlib from sys.path, so the import fails.

**Comment by PamelaM** [comment]
I've fixed the bug, but I still have a couple of things todo (see Main PR comment)

Any idea why master is failing `make lint`?
```
(agc)PamsMac:importanize pam$ make lint
flake8 importanize tests
importanize/formatters.py:176:17: W503 line break before binary operator
importanize/formatters.py:187:17: W503 line break before binary operator
importanize/groups.py:153:9: E731 do not assign a lambda expression, use a def
importanize/main.py:51:9: W503 line break before binary operator
importanize/main.py:52:9: W503 line break before binary operator
importanize/main.py:175:14: W503 line break before binary operator
importanize/main.py:176:14: W503 line break before binary operator
importanize/main.py:179:14: W503 line break before binary operator
importanize/main.py:180:14: W503 line break before binary operator
importanize/parser.py:235:5: E731 do not assign a lambda expression, use a def
importanize/parser.py:236:5: E731 do not assign a lambda expression, use a def
importanize/parser.py:237:5: E731 do not assign a lambda expression, use a def
tests/test_statements.py:176:19: W503 line break before binary operator
tests/test_statements.py:188:13: W503 line break before binary operator
tests/test_statements.py:192:13: W503 line break before binary operator
tests/test_statements.py:197:13: W503 line break before binary operator
make: *** [lint] Error 1
```

**Comment by miki725** [comment]
sorry for a late response

the lint breaks with those since when last I worked on `importanize`, those were not enforced by flake8. in most recent version they are enforced now. Ill probably fix them sometime this week.

I see you made some process on the bug! Once flake8 is fixed and I have time Ill try to play with the PR but so far looks good. Thanks for contribution!

**Comment by miki725** [comment]
@PamelaM hey. just finally got around to fixing the flake8 violations in https://github.com/miki725/importanize/commit/df96bcbbf67736b3d1254f46603a79ccc92c888b. If you want, you can pull from latest master which will fix most of the `make lint` violations

**Comment by PamelaM** [comment]
@miki725 Sorry I took so long to get back to this, but I've merged your master in and it looks great

**Comment by miki725** [comment]
sorry for the delay. Thanks @PamelaM

**Pull request merged**